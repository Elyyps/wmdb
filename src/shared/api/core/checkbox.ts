export interface ICheckbox {
  count?: number;
  isChecked?: boolean;
  label: string;
  name: string;
}

export interface ICheckboxUnique extends ICheckbox{
  id: number;
}
