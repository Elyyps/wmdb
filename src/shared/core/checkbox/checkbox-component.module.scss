.checkbox {
  -webkit-appearance: none;
  background-color: $global-background;
  border: 1px solid $gray-lighter;
  cursor: pointer;

  border-radius: 4px;
  width: 14px;
  height: 14px;
  display: inline-block;
  position: relative;
  margin: 0 8px 0 0;
  -moz-user-select: none;
  -webkit-user-select: none;
  outline: none !important;
  &-label {
    font-size: 14px;
    display: flex;
    align-items: center;
    cursor: pointer;
    color: $global-color;
    &:not(:last-child) {
      margin-bottom: 2px;
    }
  }
  &-count {
    color: $gray-light;
    margin-left: 4px;
  }
  &:checked {
    border-color: $brand-primary;
    background-color: $brand-primary;
    background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNS45NjUiIGhlaWdodD0iMTMuMTQyIiB2aWV3Qm94PSIwIDAgMTUuOTY1IDEzLjE0MiI+PHRpdGxlPmNoZWNrPC90aXRsZT48cGF0aCBkPSJNLjY3NCw3LjI5Myw1LjUyLDExLjcxNiwxNS4yMTMuNjU5IiBmaWxsPSJub25lIiBzdHJva2U9IiNmZmYiIHN0cm9rZS13aWR0aD0iMiIvPjwvc3ZnPg==);
    background-position: center;
    background-size: 10px;
    background-repeat: no-repeat;
    outline: none;
  }
}
