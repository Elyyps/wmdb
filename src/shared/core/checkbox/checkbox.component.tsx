import * as React from "react";
import styles from "./checkbox-component.module.scss";

interface ICheckboxProps {
  count?: number;
  id?: any;
  isChecked?: any;
  label?: string;
  name?: any;
  onChange?: any;
  value?: any;
}

const Checkbox = ({ name, label, id, onChange, isChecked, count, value }: ICheckboxProps) => (
  <React.Fragment>
    <label htmlFor={name} className={styles["checkbox-label"]}>
      <input
        checked={isChecked}
        type="checkbox"
        name={name}
        className={styles["checkbox"]}
        aria-checked="false"
        id={name}
        onChange={onChange}
        value={value}
      />
      {label}
      {count === 0 ? (
        <span className={styles["checkbox-count"]}>(0)</span>
      ) : (
        <span className={styles["checkbox-count"]}>({count})</span>
      )}
    </label>
  </React.Fragment>
);

export { Checkbox };
