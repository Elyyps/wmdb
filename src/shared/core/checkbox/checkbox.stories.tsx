import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { Checkbox } from "./checkbox.component";

storiesOf("Checkbox", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <Checkbox label="label" />);
