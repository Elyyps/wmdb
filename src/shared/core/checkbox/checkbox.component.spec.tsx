import React from "react";
import { shallow } from "enzyme";
import { Checkbox } from "./checkbox.component";

describe("[Label]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<Checkbox label="Label" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
