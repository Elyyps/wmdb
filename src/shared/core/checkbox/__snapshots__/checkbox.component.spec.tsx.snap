// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Label] should render component without crashing 1`] = `
<Fragment>
  <label
    className="checkbox-label"
  >
    <input
      aria-checked="false"
      className="checkbox"
      type="checkbox"
    />
    Label
  </label>
</Fragment>
`;
